### PR TITLE
YLP-2724 Add isoWeekDay field for all data types to allow filtering on week days

### PR DESCRIPTION
**Pull request by ydrisR**
## Checklist
- [x] PR name contains the Jira ticket number
- [x] Changelog updated
- [x] Code tested (unit tests and integration tests)
- [x] No console errors/warnings in integration tests
- [x] No translation files updated (except for English files)
- [x] Code formatted on every file
- [x] Copyright added on every file
- [x] End-to-end tests successful on CI

### CODE REVIEW

**Comment by ydrisR** [comment]
Linked to https://github.com/mdblp/tide-whisperer-v2/pull/82 (new version of the backend is required for tests)

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ It is based on Tidepool Blip 1.27.
 - YLP-2574 Create tool tips independently of the name of the columns
 - YLP-2704 Remove settings logic from viz
 - YLP-2726 Cleanup tideline/viz packages
+- YLP-2724 Filter on weekday for all data types where the information (weekday) is available
```
Comment (TimCanu) [inline CHANGELOG.md:10]:
```suggestion
- YLP-2724 Filter on weekday for all data types where the information (weekday) is available
- YLP-2726 Cleanup tideline/viz packages
```

File: packages/blip/app/components/chart/trendssubnav.js [hunk 2]
```diff
@@ -202,8 +202,10 @@ class TrendsSubNav extends React.Component {
       'inactive': !this.props.activeDays[day]
     }) + ' ' + day
 
+    const dayFilterId = `day-filter-${day}`
+
     return (
-      <a className={dayLinkClass} key={day} onClick={this.props.onClickDay(day)}>{this.renderDayAbbrev(day)}</a>
+      <a data-testid={dayFilterId} className={dayLinkClass} key={day} onClick={this.props.onClickDay(day)}>{this.renderDayAbbrev(day)}</a>
```
Comment (TimCanu) [inline packages/blip/app/components/chart/trendssubnav.js:208]: I understand why sonar isn't happy and it is symptomatic of another issue: I don't think that `<a>` elements should be used like this (see [here](https://developer.mozilla.org/fr/docs/Web/HTML/Element/a)).
Here, what we should use is [this](https://mui.com/material-ui/react-chip/) (basically, some kind of toggle)

File: packages/medical-domain/src/domains/repositories/medical/datum.service.ts [hunk 3]
```diff
@@ -110,15 +109,15 @@ const deduplicate = (data: Datum[], _opts: MedicalDataOptions): Datum[] => {
 export const filterOnDate = (data: Datum[], start: number, end: number, weekDaysFilter: WeekDaysFilter = defaultWeekDaysFilter): Datum[] => {
   return data.filter((dat: Datum) => {
     const epochStartInBounds = isEpochBetweenBounds(dat.epoch, start, end)
-    if (isBasal(dat) || isDuration(dat)) {
-      const epochEndInBounds = isEpochBetweenBounds(dat.epochEnd, start, end)
-      return epochStartInBounds || epochEndInBounds
-    }
-    if (isBg(dat)) {
+    if (dat.isoWeekday !== undefined) {
```
Comment (TimCanu) [inline packages/medical-domain/src/domains/repositories/medical/datum.service.ts:112]:
```suggestion
   if (dat.isoWeekday && !weekDaysFilter[dat.isoWeekday]) {
        return false
    }
```
(I couldn't make a suggestion with the line 113-116 included, but that's the spirit)

File: packages/yourloops/test/integration/pages/patient-view/trends/trends-view.common.spec.tsx [hunk 4]
```diff
@@ -84,11 +91,29 @@ describe('Trends view for anyone', () => {
       // Check Layout
       checkTrendsLayout()
       await checkTrendsBolusAndCarbsAverage()
+    })
 
-      await userEvent.click(screen.getByTestId('button-nav-back'))
+    it('should filter the data on weekday if the user removes one day', async () => {
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/patient-view/trends/trends-view.common.spec.tsx:96]: The 2 newly added tests could be added to the one above.
For various reason:
   - Run faster (no need to initialize the page many times)
   - Closer to user behaviour/experience (a user makes many actions one after the others, he does not refresh the page between each action)
   - Avoid asserting many times the same things
   - More [here](https://kentcdodds.com/blog/write-fewer-longer-tests)

We have some kind of architecture for this, I can show it to you :slightly_smiling_face:

Comment (ydrisR) [reply packages/yourloops/test/integration/pages/patient-view/trends/trends-view.common.spec.tsx]: OK

**Review by TimCanu — COMMENTED** [review]
Expect the css, it is all good. Ping me when you are available so that I can show you how to handle it :wink:

**Comment by TimCanu** [comment]
(E2E won't work anymore. Regression test will fail and will need to be regenerated. It might not work well on MAC... So I can do it if necessary)

**Review by TimCanu — APPROVED** [review]
To be merged at the same time as yourloops PR

**Pull request merged**